server/src/main/java/org/opensearch/index/translog/transfer/TranslogTransferMetadata.java

```diff
@@ -46,12 +47,27 @@ public class TranslogTransferMetadata {
 
     private final long createdAt;
 
+    private final String nodeId;
+
+    public TranslogTransferMetadata(long primaryTerm, long generation, long minTranslogGeneration, int count, String nodeId) {
+        this.primaryTerm = primaryTerm;
+        this.generation = generation;
+        this.minTranslogGeneration = minTranslogGeneration;
+        this.count = count;
+        this.createdAt = System.currentTimeMillis();
+        this.nodeId = nodeId;
+    }
+
+    /*
+    Used only at the time of download . Since details are read from content , nodeId is not available
+     */
     public TranslogTransferMetadata(long primaryTerm, long generation, long minTranslogGeneration, int count) {
         this.primaryTerm = primaryTerm;
         this.generation = generation;
         this.minTranslogGeneration = minTranslogGeneration;
         this.count = count;
         this.createdAt = System.currentTimeMillis();
+        this.nodeId = "";
```

Comment:
can we call the other constructor with the given args and nodeId = ""?

Reply:
done.